### ISSUE TITLE
Deploy doesn't send the project contents over sometimes

### ISSUE THREAD

**Issue by johnnyman727** [tessel/t2-cli#336]
I attempted to deploy the blinky example:
```
➜  t2-demo  t2 run index.js
INFO Looking for your Tessel...
INFO Connected to Harold over LAN
INFO Writing index.js to RAM on Harold (3.072 kB)...
INFO Deployed.
INFO Running index.js...
module.js:334
    throw err;
          ^
Error: Cannot find module '/tmp/remote-script/index.js'
    at Function.Module._resolveFilename (module.js:332:15)
    at Function.Module._load (module.js:282:25)
    at Function.Module.runMain (module.js:467:10)
    at startup (node.js:117:18)
    at node.js:948:3
INFO Stopping script...
```

When I checked the T2 directly, there `/tmp/remote-script` existed but had no file contents.

**Comment by johnnyman727** [comment]
I then immediately experienced #319.

**Comment by johnnyman727** [comment]
Here is a clue - it looks like the code starts to untar the contents of the project before the process that creates the folder has been completed:

```
➜  t2-demo  DEBUG=deploy t2 run index.js --timeout 3
INFO Looking for your Tessel...
INFO Connected to Harold over USB
  deploy Executing command: +0ms stopRunningScript
  deploy Command: +64ms stopRunningScript Completed
  deploy Executing command: +2ms deleteFolder
  deploy Command: +11ms deleteFolder Completed
  deploy Executing command: +1ms createFolder
  deploy Beginning processing of untarring stdin +1ms
  deploy Command: +36ms createFolder Completed
INFO Writing index.js to RAM on Harold (3.072 kB)...
  deploy Process opened. Writing stdin +1ms
INFO Deployed.
INFO Running index.js...
I'm blinking! (Press CTRL + C to stop)
```

**Comment by johnnyman727** [comment]
Fixed with #337.
